subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/video/VideoCapturer.kt

```diff
@@ -11,77 +11,105 @@ import com.avito.android.util.executeMethod
     }
 
     @SuppressLint("LogNotTimber")
-    override fun stop(): File? {
-        if (state !is State.Recording) {
-            throw RuntimeException("Unable to stop recording because recording process hasn't started yet")
-        }
-
-        val videoFile = (state as State.Recording).video
-        val outputFile = (state as State.Recording).output
-
-        executeRecorderCommand("stop", outputFile)
-
-        try {
-            waitForVideoSaving(videoFile)
-        } catch (t: Throwable) {
-            if (outputFile.exists()) {
-                Log.v(
-                    TAG,
-                    "Something went wrong during test recording. Output: ${outputFile.readText()}. Stacktrace: ${t}"
-                )
-            }
-
-            if (videoFile.exists()) {
-                videoFile.delete()
+    @Synchronized
+    override fun stop(): VideoCapturer.RecordResult {
+        val state = state
+        val result = when {
+            state is State.Recording -> {
+                val (videoFile, outputFile) = state
+                val result = try {
+                    executeRecorderCommand("stop", outputFile)
+                    waitForVideoSaving(videoFile)
+                    VideoCapturer.RecordResult.Success(videoFile)
+                } catch (t: Throwable) {
+                    if (videoFile.exists()) {
+                        videoFile.delete()
+                    }
+                    val output = if (outputFile.exists()) {
+                        outputFile.readText().also {
+                            outputFile.delete()
+                        }
+                    } else {
+                        "empty"
+                    }
+                    VideoCapturer.RecordResult.Error("Failed when stopping video record. Output: $output", t)
+                }
+                this.state = State.Idling
+                result
             }
-
-            return null
+            else -> VideoCapturer.RecordResult.Error("Can't stop capture. Capturer isn't recording")
         }
-
-        state = State.Idling
-
-        return videoFile
+        return result
     }
 
+    @Synchronized
     override fun abort() {
-        if (state is State.Recording) {
-            val videoFile = (state as State.Recording).video
-            val outputFile = (state as State.Recording).output
-
-            executeRecorderCommand("abort", outputFile)
-
-            if (videoFile.exists()) {
-                videoFile.delete()
+        val state = state
+        when {
+            state is State.Recording -> {
+                val (videoFile, outputFile) = state
+
+                try {
+                    executeRecorderCommand("abort", outputFile)
+                } catch (t: Throwable) {
+                    Log.w(TAG, "Can't abort capture", t)
+                } finally {
+                    if (videoFile.exists()) {
+                        videoFile.delete()
+                    }
```

Comment:
Should we clean `outputFile` here like in `stop()`?

Reply:
We rewrite output file every time when invoke `start` and I decided to don't clear it